blocks/bricks/base.py

```diff
@@ -21,6 +23,51 @@ def create_unbound_method(func, cls):
 property_ = property
 
 
+class Parameters(MutableSequence):
+    """Behaves exactly like a list, but annotates the variables."""
+    def __init__(self, brick, params):
+        self.brick = brick
+        self._params = []
+        for param in params:
+            self.append(param)
+
+    def __repr__(self):
+        return repr(self._params)
+
+    def __getitem__(self, key):
+        return self._params.__getitem__(key)
+
+    def _annotate(self, value):
+        """Annotates the variable.
+
+        Raises
+        ------
+        ValueError
+            If the parameter isn't a shared variable or ``None``.
+
+        """
+        if is_shared_variable(value):
+            VariableRole.add_role(value, VariableRole.PARAMETER)
+            annotations = getattr(value.tag, 'annotations', []) + [self.brick]
+            value.tag.annotations = annotations
+        elif value is not None:
```

Comment:
Wait a minute, I remember you wanted parameter substitution to be possible, like `decoder.W  = encoder.W.T`, didn't you?